src/CoreWCF.WebHttp/src/CoreWCF/UriTemplate.cs

```diff
@@ -0,0 +1,1799 @@
+                        "We have moved the parameter index in a literal binding");
+                    Fx.Assert(segmentIndex < _segments.Count,
+                        "Calculation of LastNonDefaultPathParameter,lastPathParameter or parameterIndex failed");
+                }
+                // We're done; skip to the beggining of the query parameters
+                parameterIndex = lastPathParameter + 1;
+            }
+            else if (_segments.Count > 0 || _wildcard != null)
+            {
+                for (int i = 0; i < _segments.Count; i++)
+                {
+                    _segments[i].Bind(parameters, ref parameterIndex, pathString);
+                }
+
+                if (_wildcard != null)
+                {
+                    _wildcard.Bind(parameters, ref parameterIndex, pathString);
+                }
+            }
+
+            if (IgnoreTrailingSlash && (pathString[pathString.Length - 1] == '/'))
+            {
+                pathString.Remove(pathString.Length - 1, 1);
+            }
+
+            result.Path = pathString.ToString();
+            // Binding the query :
+            if ((_queries.Count != 0) || (extraQueryParameters != null))
+            {
+                StringBuilder query = new StringBuilder("");
+                foreach (string key in _queries.Keys)
+                {
+                    _queries[key].Bind(key, parameters, ref parameterIndex, query);
+                }
+
+                if (extraQueryParameters != null)
+                {
+                    foreach (string key in extraQueryParameters.Keys)
+                    {
+                        if (_queries.ContainsKey(key.ToUpperInvariant()))
+                        {
+                            // This can only be if the key passed has the same name as some literal key
+                            throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgument(nameof(parameters), SR.Format(
+                                SR.UTBothLiteralAndNameValueCollectionKey, key));
+                        }
+                        string value = extraQueryParameters[key];
+                        string escapedValue = (string.IsNullOrEmpty(value) ? string.Empty : UrlUtility.UrlEncode(value, Encoding.UTF8));
+                        query.AppendFormat("&{0}={1}", UrlUtility.UrlEncode(key, Encoding.UTF8), escapedValue);
+                    }
+                }
+
+                if (query.Length != 0)
+                {
+                    query.Remove(0, 1); // remove extra leading '&'
+                }
+
+                result.Query = query.ToString();
+            }
+
+            // Adding the fragment (if needed)
+            if (_fragment != null)
+            {
+                result.Fragment = _fragment;
+            }
+
+            return result.Uri;
+        }
+        void BindTerminalDefaults(int numMatchedSegments, NameValueCollection boundParameters)
+        {
```

Comment:
Missing blank line. Also looks like there's quite a few methods missing accessibility modifiers in this file.

Reply:
This is addressed in e95ff7c922db5d4a6ee55847b763b49619f08311.